src/libmongoc/tests/json-test.c

```diff
@@ -1360,6 +1379,15 @@ set_uri_opts_from_bson (mongoc_uri_t *uri, const bson_t *opts)
             bson_lookup_write_concern (opts, "writeConcern");
          mongoc_uri_set_write_concern (uri, wc);
          mongoc_write_concern_destroy (wc);
+      } else if (!strcmp (bson_iter_key (&iter), "appname")) {
+         mongoc_uri_set_option_as_utf8 (
+            uri, MONGOC_URI_APPNAME, bson_iter_utf8 (&iter, NULL));
+      } else if (!strcmp (bson_iter_key (&iter), "connectTimeoutMS")) {
+         mongoc_uri_set_option_as_int32 (
+            uri, MONGOC_URI_CONNECTTIMEOUTMS, bson_iter_int32 (&iter));
+      } else if (!strcmp (bson_iter_key (&iter), "serverSelectionTimeoutMS")) {
+         mongoc_uri_set_option_as_int32 (
+            uri, MONGOC_URI_SERVERSELECTIONTIMEOUTMS, bson_iter_int32 (&iter));
```

Comment:
Would it make sense to generalize all of this logic, as we do in PHPC? See: [`php_phongo_apply_options_to_uri`](https://github.com/mongodb/mongo-php-driver/blob/41a96ba0e864450465f1ef41501ea3891cb72c83/php_phongo.c#L1478). Only RC, RP, and WC are handled separately, since those have unique setters. For everything else, we use methods like `mongoc_uri_option_is_int32` and `mongoc_uri_set_option_as_int32`.

That said, perhaps this isn't a concern since you do error on any unsupported option. That removes the risk of a test adding a new option and it being ignored by the libmongoc test runner.

Not suggesting this be addressed in this PR, but perhaps it warrants a todo item (possible `new-eng` work).